File: 2021/python/day1.py

# PART I
def part1():
  previous = None
  counter = 0

  with open("1_input.txt") as f:
    data = [int(line.strip()) for line in f.readlines()]

    for el in data:
      if previous == None:
        previous = el
        continue

      if el > previous:
        counter += 1
      
      previous = el

  return counter


# PART II
def part2():
  previous = None
  counter = 0

  with open("1_input.txt") as f:
    data = [int(line.strip()) for line in f.readlines()]

    for i in range(len(data)):
      if i >= len(data) - 2:
        break
      
      window_sum = data[i] + data[i+1] + data[i+2]

      if previous == None:
        previous = window_sum
        continue

      if window_sum > previous:
        counter += 1

      previous = window_sum

  return counter

print("Part I", part1())      
print("Part II", part2())      

File: 2021/day1.kt
import java.io.File

fun main() {
  
  val filename: String = "1_input.txt"
  val lines = File(filename).readLines()
  val nums = lines.map { it.trim().toInt() }

  fun part1(): Int {
    var previous: Int? = null
    var counter = 0

    loop@ for (num in nums) {
      if (previous == null) {
        previous = num
        continue@loop
      }

      if (num > previous) counter += 1

      previous = num
    }
    return counter
  }

  fun part2(): Int {
    var previous: Int? = null
    var counter = 0

    loop@ for (i in 0 until nums.size - 2) {
      val windowSum: Int = nums[i] + nums[i+1] + nums[i+2]

      if (previous == null) {
        previous = windowSum
        continue@loop
      }

      if (windowSum > previous!!) counter += 1

      previous = windowSum
    }
    return counter
  }

  println("Part I " + part1().toString())
  println("Part II " + part2().toString())
}